pgjdbc/src/main/java/org/postgresql/core/Parser.java

```diff
@@ -124,21 +130,27 @@
                 nativeQueries = new ArrayList<NativeQuery>();
               }
 
-              nativeQueries.add(new NativeQuery(nativeSql.toString(), toIntArray(bindPositions),isCurrentReWriteCompatible));
+              nativeQueries.add(new NativeQuery(nativeSql.toString(),
+                  toIntArray(bindPositions), DMLCommand.createStatementTypeInfo(
+                  current, isBatchedReWriteConfigured, isCurrentReWriteCompatible,
+                  isReturningPresent, isAutoCommit, nativeQueries.size())));
             }
             // Prepare for next query
             if (bindPositions != null) {
               bindPositions.clear();
             }
             nativeSql.setLength(0);
+            current = DMLCommandType.BLANK;
+            isReturningPresent = false;
           }
           break;
 
         case 'i':
           if (Parser.parseInsertKeyword(aChars, i)) {
```

Comment:
Jeremy,

Have a look at https://github.com/pgjdbc/pgjdbc/pull/547

I went ahead and did a bit of optimization and added tests

Dave Cramer

On 12 April 2016 at 10:02, Jeremy Whiting <notifications@github.com> wrote:

> In pgjdbc/src/main/java/org/postgresql/core/Parser.java
> <https://github.com/pgjdbc/pgjdbc/pull/491#discussion_r59380789>:
>
> >              }
> >              // Prepare for next query
> >              if (bindPositions != null) {
> >                bindPositions.clear();
> >              }
> >              nativeSql.setLength(0);
> > +            current = DMLCommandType.BLANK;
> > +            isReturningPresent = false;
> >            }
> >            break;
> >
> >          case 'i':
> >            if (Parser.parseInsertKeyword(aChars, i)) {
>
> yes good point. I'll re-factor the switch to convert the char to upper
> case. Which will remove the duplicate lower case block.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/pgjdbc/pgjdbc/pull/491/files/8d299ed8dce9fff0f298827a19688bc4f93faaac..067a7b85dba05ed8a8d0d1b1b00a7af91ddaa818#r59380789>
>